File: ELEC881_Copilot/src/Function Name Masked/Python/q32-concatenated-words.py
class ab(object):
    def abc(self, words):
        words = sorted(words, key=lambda x: len(x))
        words_set = set(words)
        res = []
        for word in words:
            words_set.remove(word)
            if self.dfs(word, words_set):
                res.append(word)
            words_set.add(word)
        return res
    
    def dfs(self, word, words_set):
        if word in words_set:
            return True
        for i in range(1, len(word)):
            if word[:i] in words_set and self.dfs(word[i:], words_set):
                return True
        return False

File: ELEC881_Tabnine/src/No Description Prompt/Java script (JS)/q32-concatenated-words.js
/**
 * @param {string[]} words
 * @return {string[]}
 */
var findAllConcatenatedWordsInADict = function(words) {
    const set = new Set(words);
    const res = [];

    for (let word of words) {
        if (dfs(word, 0, set, 0)) {
            res.push(word);
        }
    }

    return res.sort((a, b) => a.length - b.length);
};

function dfs(word, start, set, count) {
    if (count === 2) {
        return true;
    }

    for (let i = start; i < word.length; i++) {
        const sub = word.slice(start, i + 1);
        if (set.has(sub) && dfs(word, i + 1, set, count + 1)) {
            return true;
        }
    }

    return false;
}